pyccl/nl_pt/ept.py

```diff
@@ -0,0 +1,700 @@
+        self._pk_alias = {
+            'm:m': 'm:m', 'm:b1': 'm:m', 'm:b2': 'm:b2',
+            'm:b3nl': 'm:b3nl', 'm:bs': 'm:bs', 'm:bk2': 'm:bk2',
+            'm:c1': 'm:m', 'm:c2': 'm:c2', 'm:cdelta': 'm:cdelta',
+            'b1:b1': 'm:m', 'b1:b2': 'm:b2', 'b1:b3nl': 'm:b3nl',
+            'b1:bs': 'm:bs', 'b1:bk2': 'm:bk2', 'b1:c1': 'm:m',
+            'b1:c2': 'm:c2', 'b1:cdelta': 'm:cdelta', 'b2:b2': 'b2:b2',
+            'b2:b3nl': 'zero', 'b2:bs': 'b2:bs', 'b2:bk2': 'zero',
+            'b2:c1': 'zero', 'b2:c2': 'zero', 'b2:cdelta': 'zero',
+            'b3nl:b3nl': 'zero', 'b3nl:bs': 'zero',
+            'b3nl:bk2': 'zero', 'b3nl:c1': 'zero', 'b3nl:c2':
+            'zero', 'b3nl:cdelta': 'zero', 'bs:bs': 'bs:bs',
+            'bs:bk2': 'zero', 'bs:c1': 'zero', 'bs:c2': 'zero',
+            'bs:cdelta': 'zero', 'bk2:bk2': 'zero', 'bk2:c1': 'zero',
+            'bk2:c2': 'zero', 'bk2:cdelta': 'zero', 'c1:c1': 'm:m',
+            'c1:c2': 'm:c2', 'c1:cdelta': 'm:cdelta', 'c2:c2': 'c2:c2',
+            'c2:cdelta': 'c2:cdelta', 'cdelta:cdelta': 'cdelta:cdelta'}
+        # All valid Pk pair labels
+        self._pk_valid = list(self._pk_alias.keys())
+        # List of Pk2Ds to fill out
+        self._pk2d_temp = {}
+
+    def _check_pt_init(self):
+        if self._pt_init:
+            return
+        raise RuntimeError("PT templates have not been initialised "
+                           "for this calculator. Please do so using "
+                           "`update_ingredients`.")
+
+    @unlock_instance
+    def update_ingredients(self, cosmo):
+        """ Update the internal PT arrays.
+
+        Args:
+            cosmo (:class:`~pyccl.core.Cosmology`): a `Cosmology` object.
+        """
+        pklz0 = cosmo.linear_matter_power(self.k_s, 1.0)
+        g = cosmo.growth_factor(self.a_s)
+        self._g4 = g**4
+
+        # Galaxy clustering templates
+        if self.with_NC:
+            self.dd_bias = self.pt.one_loop_dd_bias_b3nl(
+                pklz0, P_window=self.fastpt_par['P_window'],
+                C_window=self.fastpt_par['C_window'])
+            self.one_loop_dd = self.dd_bias[0:1]
+            self.with_matter_1loop = True
+        elif self.with_matter_1loop:  # Only 1-loop matter needed
+            self.one_loop_dd = self.pt.one_loop_dd(
+                pklz0, P_window=self.fastpt_par['P_window'],
+                C_window=self.fastpt_par['C_window'])
+
+        # Intrinsic alignment templates
+        if self.with_IA:
+            self.ia_ta = self.pt.IA_ta(
+                pklz0, P_window=self.fastpt_par['P_window'],
+                C_window=self.fastpt_par['C_window'])
+            self.ia_tt = self.pt.IA_tt(
+                pklz0, P_window=self.fastpt_par['P_window'],
+                C_window=self.fastpt_par['C_window'])
+            self.ia_mix = self.pt.IA_mix(
+                pklz0, P_window=self.fastpt_par['P_window'],
+                C_window=self.fastpt_par['C_window'])
```

Comment:
Since they all use the same kwargs, instead of copying+pasting the same thing 3 times you can do
```python
kwargs = {"P": pklz0, "P_window": self.fastpt_par["P_window"], "C_window": self.fastpt_par["C_window"]}
self.ia_ta = self.pt.IA_ta(**kwargs)
self.ia_tt = ...
```

Reply:
Done